src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,143 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head; // The first node of the linked list
+    private Node<T> tail; // The last node of the linked list
```

Comment:
If you think the words "first" and "last" better reflect the sense of these fields then "head" and "tail" - use them for field names. But we do not need such comments)